h2/src/main/org/h2/mvstore/DataUtils.java

```diff
@@ -681,7 +681,14 @@ private static int parseMapValue(StringBuilder buff, String s, int i, int size)
      * @throws IllegalStateException if parsing failed
      */
     public static HashMap<String, String> parseChecksummedMap(byte[] bytes) {
-        String s = new String(bytes, StandardCharsets.ISO_8859_1).trim();
+        int start = 0, end = bytes.length;
+        while (start < end && bytes[start] <= ' ') {
+            start++;
+        }
+        while (start < end && bytes[end - 1] <= ' ') {
+            end--;
+        }
```

Comment:
can probably trim NUL bytes off the end too?

Reply:
This code already trims `(char) 0` from the end, not only spaces.

Reply:
I mean `String.trim() ` that was used before trims NUL from the both ends and my code do exactly the same work.